src/runtime_memory.c

```diff
@@ -141,21 +141,39 @@ void oc_wasm_list_push_back(oc_wasm_list* list, oc_wasm_list_elt* elt)
     list->last = eltAddr;
 }
 
+oc_wasm_str8 oc_wasm_str8_from_native(oc_wasm_addr arena, oc_str8 nativeString)
+{
+    oc_wasm_str8 wasmString;
+    wasmString.len = nativeString.len;
+    wasmString.ptr = oc_wasm_arena_push(arena, nativeString.len);
```

Comment:
I generally prefer designated init:
```
oc_wasm_str8 wasmString = {
    .len = nativeString.len,
    .ptr = oc_wasm_arena_push(arena, nativeString.len),
};
```
This ensures whatever field we add gets zero initialized, which makes it either work or crash in obvious ways compared to an uninitialized field